src/services/coingecko.ts

```diff
@@ -33,24 +33,33 @@ export function getOriginalAddress(chainId: number, address: string) {
     return;
   }
   const map = {
+    // WETH
     '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2': {
-      42: '0xdfcea9088c8a88a76ff74892c1457c17dfeef9c1'
+      42: '0xdfcea9088c8a88a76ff74892c1457c17dfeef9c1',
+      137: '0x7ceB23fD6bC0adD59E62ac25578270cFf1b9f619'
     },
+    // WBTC
     '0x2260fac5e5542a773aa44fbcfedf7c193bc2c599': {
       42: '0x1c8e3bcb3378a443cc591f154c5ce0ebb4da9648'
     },
+    // BAL
     '0xba100000625a3754423978a60c9317c58a424e3d': {
-      42: '0x41286bb1d3e870f3f750eb7e1c25d7e48c8a1ac7'
+      42: '0x41286bb1d3e870f3f750eb7e1c25d7e48c8a1ac7',
+      137: '0x9a71012b13ca4d3d0cdc72a177df3ef03b0e76a3'
```

Comment:
I've added polygon addresses only to the tokens that are defined in `tokens.ts` (`WETH` and `BAL`).
If I understood this function correctly it will map the original Mainnet addresses to the correct network address tokens.